src/roiextractors/multiimagingextractor.py

```diff
@@ -53,17 +53,23 @@ def _check_consistency_between_imaging_extractors(self):
                 len(unique_values) == 1
             ), f"{property_message} is not consistent over the files (found {unique_values})."
 
-    @check_get_frames_args
     def get_frames(self, frame_idxs: ArrayType, channel: Optional[int] = 0) -> NumpyArray:
-        extractor_indices = np.searchsorted(self._end_frames, frame_idxs, side="right")
-        # Match frame_idxs to imaging extractors
-        extractors_dict = defaultdict(list)
-        for extractor_index, frame_index in zip(extractor_indices, frame_idxs):
-            extractors_dict[extractor_index].append(frame_index)
+        if isinstance(frame_idxs, (int, np.integer)):
+            frame_idxs = [frame_idxs]
+        frame_idxs = np.array(frame_idxs)
+        if all([isinstance(frame_idx, Iterable) for frame_idx in frame_idxs]):
+            frame_idxs = frame_idxs.flatten()
```

Comment:
I'd still say supporting it in this way is unnecessary. If you remove it you can go back to just using the decorator